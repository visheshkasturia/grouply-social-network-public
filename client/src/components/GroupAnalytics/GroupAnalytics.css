.center {
    padding: 10px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
}

.statistics-group-container {
    display: grid;
    /* grid-template-columns: 1; */
    grid-template-rows: 5;
    border: solid 4px black;
    border-radius: 10px;
    padding: 20px;
    height: 200px;
    width: 250px;
    background: #f0e7e6;
    margin: 10px;
}

.statistics-group-name {
    grid-row: 1/ span 1;
}