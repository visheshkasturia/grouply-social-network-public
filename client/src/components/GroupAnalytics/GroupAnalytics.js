import React from 'react';
import './GroupAnalytics.css';
/* eslint-disable */
function GroupAnalytics({ group, user }) {

  return (
    <div className="statistics-group-container">
      <div className="statistics-group-name"> <b>Group Name:</b> {group.groupName}</div>
      {/* <div className ="explore-group-description">tags:{group.tags[0]}</div> */}
      <div className ="statistics-group-tags"> 
        <b>Tags:</b>
        <ul>
        {group.tags.map((t, i) => (
          <li key={i}>{t}</li>))}
        </ul>
      </div>
      <div className="statistics-group-members"><b>Number of Members:</b> {group.allMembers.length }</div>
      <div className="statistics-group-posts"> <b>Number of Posts:</b>  {group.allPosts.length }</div>
      <div className ="statistics-group-admins"> 
        <b>Admins:</b>
        <ul>
        {group.admins.map((t, i) => (
          <li key={i}>{t}</li>))}
        </ul>
      </div>
      
    </div>
  )
}

export default GroupAnalytics
