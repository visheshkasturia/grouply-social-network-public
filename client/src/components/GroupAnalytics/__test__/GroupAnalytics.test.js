import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import GroupAnalytics from '../GroupAnalytics'

const admin = 'abc'
const member = 'test1'
const post = '1'
const tag = 'tag'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin],
  allMembers: [member],
  allPosts: [post],
  tags: [tag]
}

it('GroupAdmins renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GroupAnalytics group={group} />, div);
});

it('GroupAdmins matches snapshot', () => {
  const tree = renderer.create(<GroupAnalytics group={group} />).toJSON();
  expect(tree).toMatchSnapshot();
});
