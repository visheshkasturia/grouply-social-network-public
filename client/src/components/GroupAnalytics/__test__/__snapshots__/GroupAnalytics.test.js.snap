// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupAdmins matches snapshot 1`] = `
<div
  className="statistics-group-container"
>
  <div
    className="statistics-group-name"
  >
     
    <b>
      Group Name:
    </b>
     
    test
  </div>
  <div
    className="statistics-group-tags"
  >
    <b>
      Tags:
    </b>
    <ul>
      <li>
        tag
      </li>
    </ul>
  </div>
  <div
    className="statistics-group-members"
  >
    <b>
      Number of Members:
    </b>
     
    1
  </div>
  <div
    className="statistics-group-posts"
  >
     
    <b>
      Number of Posts:
    </b>
      
    1
  </div>
  <div
    className="statistics-group-admins"
  >
    <b>
      Admins:
    </b>
    <ul>
      <li>
        abc
      </li>
    </ul>
  </div>
</div>
`;
